transport/src/main/java/io/netty/channel/EventLoopGroups.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright 2016 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.netty.channel;
+
+import io.netty.util.internal.SystemPropertyUtil;
+import io.netty.util.internal.logging.InternalLogger;
+import io.netty.util.internal.logging.InternalLoggerFactory;
+
+import java.lang.reflect.Constructor;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.ThreadFactory;
+
+/**
+ * {@link EventLoopGroup} utilities.
+ */
+public final class EventLoopGroups {
+    private static final InternalLogger logger = InternalLoggerFactory.getInstance(EventLoopGroups.class);
+
+    protected static final String CHANNEL_PACKAGE = SystemPropertyUtil.get(
+            "io.netty.packagePrefix", EventLoopGroups.class.getPackage().getName());
+
+    private static final List<String> STANDARD_IMPLEMENTATIONS = Collections.unmodifiableList(Arrays.asList(
+            "nio", "oio"));
+
+    private static final List<String> NATIVE_IMPLEMENTATIONS = Collections.unmodifiableList(Arrays.asList(
+            "epoll"
```

Comment:
Currently "native" implementations and "standard" implementations are decoupled. The `transport` module does not know about the "native" modules. Is there a way we can preserve this relationship and avoid adding a cross dependency (even if its implicit through string manipulation)? For example can the "native" modules load/register them selves into these maps using runtime type information?